File: vscodeExt/167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input array is sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0 
        r = len(numbers) - 1
        while l < r:
            if numbers[l]+numbers[r] == target:
                return [l+1,r+1]
            elif numbers[l]+numbers[r] < target:
                l += 1
            else:
                r -= 1        
        
# @lc code=end

File: vscodeExt/167.two-sum-ii-input-array-is-sorted.java
/*
 * @lc app=leetcode id=167 lang=java
 *
 * [167] Two Sum II - Input array is sorted
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int begin =0;
        int end = numbers.length-1;
        while(begin<end){
            if(numbers[begin]+ numbers[end]==target){
                break;
            } else if(numbers[begin]+ numbers[end]<target) {
                begin+=1;
            } else {
                end -=1;
            }

        }
        return new int[]{begin+1,end+1};
    }
}
// @lc code=end

